### PR TITLE
Callout max size

### PR DESCRIPTION
**Pull request by shubham7109**
PR to calculate the max-content size of the Callout container. For testing apply insets to the MapView to verify the max content size:
```kotlin
@Composable
fun MainScreen(viewModel: MapViewModel) {

    val mapPoint = viewModel.mapPoint.collectAsState().value

    MapView(
        modifier = Modifier.fillMaxSize(),
        arcGISMap = viewModel.arcGISMap,
        onSingleTapConfirmed = viewModel::setMapPoint,
        insets = PaddingValues(50.dp), // SET INSETS
        content = if (mapPoint != null) {
            {
                Callout(location = mapPoint) {
                    // ADD LARGE CALLOUT CONTENT
                    Text(
                        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Euismod quis viverra nibh cras pulvinar. Turpis massa tincidunt dui ut ornare lectus sit amet est. Nibh mauris cursus mattis molestie a iaculis at erat. Pharetra vel turpis nunc eget. Habitasse platea dictumst quisque sagittis.",
                        color = Color.Green
                    )
                }
            }
        } else {
            null
        }
    )
}

```

### CODE REVIEW

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt [hunk 1]
```diff
@@ -38,8 +39,11 @@ import androidx.compose.ui.graphics.Path
 import androidx.compose.ui.graphics.drawscope.Fill
 import androidx.compose.ui.graphics.drawscope.Stroke
 import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.platform.LocalConfiguration
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt:42]: Some unused imports here that could be cleaned up

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt [hunk 2]
```diff
@@ -222,6 +233,35 @@ private fun DoubleXY.rotate(rotateByAngle: Double, center: DoubleXY = DoubleXY.z
...
+private fun calloutContentMaxSize(
+    geoView: GeoView,
+    density: Density,
+    displayMetrics: DisplayMetrics
+): DpSize {
+    // Start by getting height & width of GeoView
+    var maxHeightForGeoView = geoView.height
+    var maxWidthForGeoView = geoView.width
+    if (maxHeightForGeoView == 0 || maxWidthForGeoView == 0) {
+        // Use width of display if view width not available yet (as happens once, before it is measured)
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt:249]: This comment is not valid for height. Also I am not sure if we should be affecting both values if either the Height or width is zero. Can we separate them out in their respective if blocks. Or will it be better to affect them both if one is unknown ?

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt [hunk 3]
```diff
@@ -222,6 +233,35 @@ private fun DoubleXY.rotate(rotateByAngle: Double, center: DoubleXY = DoubleXY.z
...
+        maxHeightForGeoView = displayMetrics.heightPixels
+        maxWidthForGeoView = displayMetrics.widthPixels
+    }
+    // if we have valid insets set on the MapView, we deduct the maxHeightForGeoView & maxWidthForMapView  by the specified inset sizes
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt:253]:
```suggestion
    // if we have valid insets set on the MapView, we deduct the maxHeightForGeoView & maxWidthForMapView by the specified inset sizes
```

File: toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt [hunk 4]
```diff
@@ -222,6 +233,35 @@ private fun DoubleXY.rotate(rotateByAngle: Double, center: DoubleXY = DoubleXY.z
     return DoubleXY(x2 + center.x, y2 + center.y)
 }
 
+/**
+ * Determines the max content size of the Callout container factoring in the Insets set on the GeoView.
```
Comment (puneet-pdx) [inline toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/Callout.kt:237]:
```suggestion
 * Determines the max content size of the Callout container factoring in the size of the GeoView and the Insets set on the GeoView.
```

**Review by puneet-pdx — COMMENTED** [review]
@shubham7109 some questions/comments

**Review by sorenoid — APPROVED** [review]

**Pull request merged**